src/command_modules/azure-cli-advisor/azure/cli/command_modules/advisor/_params.py

```diff
@@ -0,0 +1,24 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+from azure.cli.core.commands import register_cli_argument
+
+register_cli_argument('advisor recommendation generate', 'timeout', options_list=('--timeout', '-t'), required=False, help='Timeout in seconds')
```

Comment:
Include `type=int`. This will automatically handle checking that the type is an int and will convert it automatically as well then in your command function that uses `timeout`, you'll know that you will always have an int.